src/ics/icalendar.py

```diff
@@ -118,7 +118,7 @@ def normalize(self, normalization, *args, **kwargs):
                 raise ValueError("can't pass args or kwargs when a complete Normalization is given")
         else:
             normalization = Normalization(normalization, *args, **kwargs)
-        self.events = [normalization.normalize(e) for e in self.events]
+        self.events = [e if e.all_day else normalization.normalize(e) for e in self.events]
         self.todos = [normalization.normalize(e) for e in self.todos]
```

Comment:
In particular, I am not sure if TODOs have times and could have the same all-day event+timezone normalization issue...